src/App.tsx

```diff
@@ -1,73 +1,105 @@
+    setSelectedPerson(null);
+  };
+
+  const handleInputFocus = () => {
+    setIsDropdownActive(true);
+  };
+
+  const reset = () => {
+    setQuery('');
+    setSelectedPerson(null);
+    setAppliedQuery('');
+  };
+
+  const handleUserSelect = (user: Person) => {
+    setSelectedPerson(user);
+    setQuery(user.name);
+    setAppliedQuery(user.name); // alway add deley 300
+    setIsDropdownActive(false);
+  };
+
+  const title = selectedPerson
+    ? `${selectedPerson.name} (${selectedPerson.born} - ${selectedPerson.died})`
+    : 'No selected person';
 
   return (
     <div className="container">
       <main className="section is-flex is-flex-direction-column">
         <h1 className="title" data-cy="title">
-          {`${name} (${born} - ${died})`}
+          {title}
         </h1>
 
-        <div className="dropdown is-active">
+        <div
+          className={cn('dropdown', { 'is-active': isDropdownActive })}
+          onMouseLeave={() => setIsDropdownActive(false)}
+        >
           <div className="dropdown-trigger">
             <input
+              value={query}
+              onChange={handleChange}
+              onFocus={handleInputFocus}
               type="text"
               placeholder="Enter a part of the name"
               className="input"
               data-cy="search-input"
             />
           </div>
 
-          <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
-            <div className="dropdown-content">
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Haverbeke</p>
-              </div>
+          {selectedPerson && (
+            <button onClick={reset} className="button is-danger">
+              <i className="fas fa-trash"></i>
+            </button>
+          )}
 
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Bernard Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter Antone Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Haverbeke</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-link">Pieter de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Petronella de Decker</p>
-              </div>
-
-              <div className="dropdown-item" data-cy="suggestion-item">
-                <p className="has-text-danger">Elisabeth Hercke</p>
-              </div>
-            </div>
-          </div>
+          <DropdownMenu users={filterUsers} onUserSelect={handleUserSelect} />
```

Comment:
you shouldn't show an empty dropdown list
<img width="415" alt="Screenshot 2024-04-06 at 12 43 54" src="https://github.com/mate-academy/react_autocomplete/assets/117587463/a8f31a42-dc43-4088-91d0-a32ad61dbb49">